### DISCUSSION TITLE
SQL Server relations causing queries to hit parameter limit

### DISCUSSION THREAD

**Discussion by PatricioNG** [prisma/prisma#21776] [Q&A] [answered]
### Question

I've done a bit of digging around but haven't seen this error come up a lot so wondering how others using SQL Server are working around it? Prisma very often causes SQL server to run into the parameter limit when using `includes` for related tables. ie:
```
// A whole bunch of this same error text above
[oms_picked_orders].[boxID] = @P4190) OR ([dbo].[oms_picked_orders].[orderNumber] = @P4191 AND [dbo].[oms_picked_orders].[boxID] = @P4192) OR ([dbo].[oms_picked_orders].[orderNumber] = @P4193 AND [dbo].[oms_picked_orders].[boxID] = @P4194) OR ([dbo].[oms_picked_orders].[orderNumber] = @P4195 AND [dbo].[oms_picked_orders].[boxID] = @P4196) OR ([dbo].[oms_picked_orders].[orderNumber] = @P4197 AND [dbo].[oms_picked_orders].[boxID] = @P4198))
prisma:error
Invalid `prisma.boxes.findMany()` invocation:


Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(TokenError { code: 8003, state: 1, class: 16, message: "The incoming request has too many parameters. The server supports a maximum of 2100 parameters. Reduce the number of parameters and resend the request.", server: "OMS-DEV-DB01", procedure: "", line: 1 }), transient: false })
```

It's quite frustrating as the query that's triggering this is simply a find many with some includes

```
await DBClient.OMS.boxes.findMany({
			where: {
				status: { in: Constants.OrderProcessing.Orders.STATUS_TYPES.IN_PROGRESS },
				coast: { in: warehouseIDs },
				sale_type: Constants.Enums.Boxes.SaleType.Sale,
				postdate_UTC: { gte: DateTime.now().minus({ days: 30 }).toJSDate() },
				isDistroPickable: true,
			},
			include: {
				oms_picked_orders: true,
				oms_staged_order_picking: true,
			},
		});
```

So then I have to manage either slicing up my queries completely, or approaching in different ways. I know it's a SQL Server thing, but does anyone have any advice on how they are navigating this? My ideal would be that prisma managed this in the engine, if the count is going to go over 2100 it would slice it up into batches automatically and then join the results together before returning to the user. 🤔

**Comment by janpio** [comment, marked as answer]
This looks like a bug. We generally have "chunking" logic that should split such queries so that you do not run into this limit.
Can please you open a bug report and fill all the information of the bug issue template so we can reproduce? Thank you.

**Reply by PatricioNG** [reply]
Ah thanks @janpio, glad to hear this is something that could be resolved 😄 I've created a basic reproduction here https://github.com/PatricioNG/prisma-sql-server-parameter-error/tree/main 

Will open an issue now for it 👍

**Reply by PatricioNG** [reply]
Issue is opened: https://github.com/prisma/prisma/issues/21778
